src/Symfony/Component/Messenger/Command/SetupTransportsCommand.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Messenger\Transport\SetupableTransportInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class SetupTransportsCommand extends Command
+{
+    protected static $defaultName = 'messenger:setup-transports';
+
+    private $senderLocator;
+    private $receiverNames;
+
+    public function __construct(ContainerInterface $senderLocator, array $receiverNames = [])
+    {
+        $this->senderLocator = $senderLocator;
+        $this->receiverNames = $receiverNames;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        foreach ($this->receiverNames as $id => $receiverName) {
+            $receiver = $this->senderLocator->get($id);
+            if ($receiver instanceof SetupableTransportInterface) {
+                $receiver->setup();
+                $io->success("Setup $id");
+            }
```

Comment:
I'd suggest we add a `else` with a message staying that this transport has not been set up because it's not setup-able.

Reply:
Yes!

And actually I'm not sure that Amqp should be setup-able. For example, in #30557, we need to set up new queues at runtime to support delays. We can only really set up the *main* (non-delayed queues). So, we either need to make AMQP not setup-able, or we need to explicitly say "if you use manual Amqp set up, then you can't use any delays".

Reply:
The else is there now. Let's leave it as is. If you remove auto-setup from AMQP and try to use retry, you'll get an exception... because you messed up :).